cms/djangoapps/contentstore/views/public.py

```diff
@@ -46,7 +46,7 @@ def redirect_to_lms_login_for_admin(request):
     """
     This view redirect the admin/login url to the site's login page.
     """
-    return redirect('/login?next=/admin')
+    return redirect('/login/?next=/admin')
```

Comment:
Ok, I’m confused as to what this does I guess. Shouldn’t the url patterns not be changing in this case, if they are in fact the same as before?

Reply:
After merging the content of [urls_common.py](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/user_authn/urls_common.py) into the [urls.py](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/user_authn/urls.py) file and replacing `user_authn.urls_common` with `user_authn.urls` in `cms/urls.py`, the [login endpoint of user_authn](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/user_authn/urls.py#L18) was also added to cms urls.

With this change, there was this `admin/login/` [endpoint](https://github.com/openedx/edx-platform/blob/master/cms/urls.py#L242) (admin login endpoint) that started resolving to user_authn login endpoint instead of [oauth2_urlpatterns](https://github.com/openedx/edx-platform/blob/master/cms/urls.py#L51) login endpoint and was causing error. After debugging, I found out that the `redirect('/login?next=/admin')` inside `redirect_to_lms_login_for_admin` has `/login` instead of `/login/`. 

Since the [LOGIN_URL](https://github.com/openedx/edx-platform/blob/master/cms/envs/common.py#L813) of cms also has the value `/login/`, so I replaced `/login` with `/login/` in the `redirect_to_lms_login_for_admin` to make it consistent with cms LOGIN_URL. It also fixed the issue.

Reply:
Please let me know if you think its not the right approach to resolve this.